### PR TITLE
Add concurrentqueue/1.0.1

### PR DESCRIPTION
**Pull request by yipdw**
Specify library name and version:  **concurrentqueue/1.0.1**

Library source: https://github.com/cameron314/concurrentqueue

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.  _`yapf -i conanfile.py` produces no changes._
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version. _Package builds with conan 1.22.3._
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.  _Package create and test succeeds using g++ 9.2.1 on Kubuntu 19.10 x86-64.  All hook checks pass._

### CODE REVIEW

**Comment by yipdw** [comment]
An older version of my branch had incorrect license metadata; 83c42ce3eda217cd78492e1066d0e8a0cf5ed941 corrects this.

File: recipes/moodycamel-concurrentqueue/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from conans import ConanFile, CMake, tools
+from fnmatch import fnmatch
+import os
+
+
+class MoodycamelConcurrentqueueConan(ConanFile):
+    name = "moodycamel-concurrentqueue"
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/conanfile.py]:
```suggestion
    name = "concurrentqueue"
```

Follow the original project name

File: recipes/moodycamel-concurrentqueue/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/cameron314/concurrentqueue"
+    description = "A fast multi-producer, multi-consumer lock-free concurrent queue for C++11"
+    topics = ("cpp11", "cpp14", "cpp17", "queue", "lock-free")
+    license = "BSD-2-Clause, BSL-1.0"
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/conanfile.py]:
```suggestion
    license = ["BSD-2-Clause", "BSL-1.0"]
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(PackageTest CXX)
+set(CMAKE_CXX_STANDARD 11)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(Threads REQUIRED)
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: export it in cpp_info.
```suggestion
```

File: recipes/moodycamel-concurrentqueue/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    no_copy_source = True
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/conanfile.py]:
```suggestion
    no_copy_source = True
    settings = "os"
```

File: recipes/moodycamel-concurrentqueue/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                  self._source_subfolder)
+
+    def package(self):
+        self.copy("*.h",
+                  src=os.path.join(self._source_subfolder),
+                  dst=os.path.join("include", "moodycamel"))
+        self.copy("LICENSE.md", src=self._source_subfolder, dst="licenses")
+
+    def package_id(self):
+        self.info.header_only()
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/conanfile.py]:
```suggestion
        self.info.header_only()
        
    def package_info(self):
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["pthread"]
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(PackageTest CXX)
+set(CMAKE_CXX_STANDARD 11)
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]:
```suggestion
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(Threads REQUIRED)
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS} Threads::Threads)
```
Comment (uilianries) [inline recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]:
```suggestion
set_property(TARGET example PROPERTY CXX_STANDARD 11)
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (yipdw) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: Is this possible? `find_package(Threads)` does quite a bit of target-specific work; should all of that be duplicated?

Comment (uilianries) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: Yes, if you use https://github.com/conan-io/conan-center-index/pull/1052#discussion_r389035996 approach

Comment (yipdw) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: I guess that's close enough for most platforms in use, though I would have liked a way to enable that package for the CMake generator to reuse the work it does.  I'll get that working for my targets of interest and push the result.

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt (hunk 7 above)
Comment (madebr) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: You still might want to link to `${CONAN_LIBS}`

File: recipes/moodycamel-concurrentqueue/all/test_package/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+import os
+
+class MoodycamelConcurrentqueueTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
```
Comment (madebr) [inline recipes/moodycamel-concurrentqueue/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt (hunk 7 above)
Comment (yipdw) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]: I've changed this to 

```
target_link_libraries(example ${CONAN_LIBS})
set_property(TARGET example PROPERTY CXX_STANDARD 11)
```

File: recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (yipdw) [reply recipes/moodycamel-concurrentqueue/all/test_package/CMakeLists.txt]:
```python
    def package_info(self):
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["pthread"]
```

added to `conanfile.py`.

**Comment by yipdw** [comment]
I think all requested changes have been applied.

**Comment by uilianries** [comment]
Great job @yipdw Thanks!

**Comment by yipdw** [comment]
@uilianries Thank you for your comments!

**Pull request merged**